airbyte-local-cli-nodejs/src/command.ts

```diff
@@ -0,0 +1,286 @@
+import {Command, Option} from 'commander';
+
+import {logger, parseConfigFile, updateLogLevel} from './utils';
+import {CLI_VERSION} from './version';
+
+// Cli options provided by the user
+interface CliOptions {
+  // source and destination config
+  configFile?: string;
+  wizard?: object;
+
+  // source
+  srcImage: string;
+  srcConfig?: Record<string, any>;
+  srcOutputFile?: string;
+  srcCheckConnection?: boolean;
+  srcOnly?: boolean;
+  srcPull?: boolean;
+
+  // destination
+  dstImage: string;
+  dstConfig?: Record<string, any>;
+  dstUseHostNetwork?: boolean;
+  dstOnly?: string;
+  dstPull?: boolean;
+
+  // general connector settings
+  connectionName?: string;
+  stateFile: string;
+  fullRefresh?: boolean;
+  rawMessages?: boolean;
+  keepContainers?: boolean;
+  logLevel?: string;
+
+  // logging
+  debug?: boolean;
+}
+
+// Airbyte connector source or destination config
+export interface AirbtyeConfig {
+  image: string;
+  config?: object;
+  catalog?: object;
+  dockerOptions?: string;
+}
+
+enum AirbyteConfigInputType {
+  FILE = 'file',
+  OPTION = 'option',
+}
+
+// Config that are needed for running this Airbyte Cli
+export interface FarosConfig {
+  src?: AirbtyeConfig;
+  dst?: AirbtyeConfig;
+  srcOutputFile?: string | undefined; // if srcOnly is true
+  srcInputFile?: string | undefined; // if dstOnly is true
+  srcCheckConnection?: boolean | undefined;
+  dstUseHostNetwork?: boolean | undefined;
+  srcPull?: boolean | undefined;
+  dstPull?: boolean | undefined;
+  connectionName?: string | undefined;
+  stateFile: string | undefined;
+  fullRefresh?: boolean | undefined;
+  rawMessages?: boolean | undefined;
+  keepContainers?: boolean | undefined;
+  logLevel?: string | undefined;
```

Comment:
as there's a few options are affecting each other, and we need to merge and src/dst from 3 ways (file | src/dst option | wizard).
the narrowed down one would be only these.